godo.go

```diff
@@ -162,6 +162,48 @@ func NewClient(httpClient *http.Client) *Client {
 	return c
 }
 
+// ClientOpt are options for New.
+type ClientOpt func(*Client) error
+
+// New returns a new DIgitalOcean API client instance.
+func New(httpClient *http.Client, opts ...ClientOpt) (*Client, error) {
+	if httpClient == nil {
+		httpClient = http.DefaultClient
+	}
+
+	baseURL, _ := url.Parse(defaultBaseURL)
+
+	c := &Client{client: httpClient, BaseURL: baseURL, UserAgent: userAgent}
+	for _, opt := range opts {
+		if err := opt(c); err != nil {
+			return nil, err
+		}
+	}
+
+	return c, nil
+}
+
+// BaseURL is a client option for setting the base URL.
+func BaseURL(bu string) ClientOpt {
+	return func(c *Client) error {
+		u, err := url.Parse(bu)
+		if err != nil {
+			return err
+		}
+
+		c.BaseURL = u
+		return nil
+	}
+}
+
+// UserAgent is a client option for setting the user agent.
+func UserAgent(ua string) ClientOpt {
```

Comment:
same same, `Set...`. also makes it easier to 'guess' what the options are since they all start by `Set` or some equivalent.